packages/ERTP/src/paymentLedger.js

```diff
@@ -48,8 +48,36 @@ export const makePaymentLedger = (
     throw reason;
   };
 
-  /** @type {WeakStore<Payment, Amount>} */
+  /** @type {WeakMapStore<Payment, Amount>} */
   const paymentLedger = makeScalarBigWeakMapStore('payment');
+  /** @type {WeakMapStore<Payment, SetStore<Payment>>} */
+  const paymentRecoverySets = makeScalarBigWeakMapStore('payment-recovery');
+
+  /**
+   * @param {Payment} payment
+   * @param {Amount} amount
+   * @param {SetStore<Payment>} [optRecoverySet]
+   */
+  const initPayment = (payment, amount, optRecoverySet = undefined) => {
+    if (optRecoverySet !== undefined) {
+      if (!optRecoverySet.has(payment)) {
+        optRecoverySet.add(payment);
+      }
```

Comment:
Is protecting this `.add` with an if statement useful?  It looks like just a distraction.